app/services/wallet.js

```diff
@@ -967,3 +968,65 @@ export const nodeInfo = async () => {
     throw new Error('Unable to connect to ShockAPI')
   }
 }
+/**
+ * https://api.lightning.community/#routingpolicy
+ * @typedef {object} RoutingPolicy
+ * @prop {number} time_lock_delta
+ * @prop {number} min_htlc
+ * @prop {number} fee_base_msat
+ * @prop {number} fee_rate_milli_msat
+ * @prop {number} last_update
+ * @prop {max_htlc_msat} fee_rate_milli_msat
+ * @prop {last_update} fee_rate_milli_msat
+ * @prop {boolean} disabled
+ */
+
+/**
+ * https://api.lightning.community/#grpc-response-channeledge
+ * @typedef {object} ChanInfo
+ * @prop {number} channel_id
+ * @prop {number} last_update
+ * @prop {number} capacity
+ * @prop {string} chan_point
+ * @prop {string} node1_pub
+ * @prop {string} node2_pub
+ * @prop {RoutingPolicy} node1_policy
+ * @prop {RoutingPolicy} node2_policy
+ */
+
+/**
+ * @param {string | undefined} chanID
+ * @returns {Promise<ChanInfo>}
+ */
+export const getChaninfo = async chanID => {
+  if (chanID === undefined) {
```

Comment:
If the function is not expecting chanID as undefined (because it throws an error), then change the typings from `string | undefined` to `string` and let typescript handle the cases where you might pass undefined to this function.

Also correct approach would be `typeof chanID === 'undefined'`